From a50eafd0e9edd04141e60f6a2183d7bdc7da9447 Mon Sep 17 00:00:00 2001
From: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
Date: Mon, 16 Jul 2018 11:36:59 +0200
Subject: [PATCH] PMCPESYS-51: thermal: ltq: Introduce ltq_thermal_sensor
 struct

This structure represents a sensor cell and contains sensor
specific properties.

Signed-off-by: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
---
 drivers/thermal/ltq_thermal.c | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/drivers/thermal/ltq_thermal.c b/drivers/thermal/ltq_thermal.c
index 3b1bf8aaf18a..5f1849130b87 100644
--- a/drivers/thermal/ltq_thermal.c
+++ b/drivers/thermal/ltq_thermal.c
@@ -34,6 +34,21 @@ struct ltq_thermal;
 struct ltq_thermal_tsens_data {
 	void (*init)(struct platform_device *pdev, struct ltq_thermal *p);
 	int (*get_temp)(struct ltq_thermal *p);
+	u8 sensors_count;
+};
+
+struct ltq_thermal_sensor {
+	int id;
+
+	struct thermal_zone_device *tzd;
+
+	struct ltq_thermal_tsens_data *pdata;
+
+	int temp;
+	int last_temp;
+	int emul_temp;
+
+	void *drvdata;
 };
 
 struct ltq_thermal {
@@ -43,6 +58,9 @@ struct ltq_thermal {
 	struct ltq_thermal_tsens_data *tdata;
 	struct thermal_zone_device *tzd;
 
+	struct ltq_thermal_sensor *sensors;
+	int count;
+
 	int temp;
 	int last_temp;
 
@@ -142,6 +160,7 @@ int ltq_grx500_get_temp(struct ltq_thermal *priv)
 static struct ltq_thermal_tsens_data ltq_grx500_data = {
 	.init		= ltq_grx500_init,
 	.get_temp	= ltq_grx500_get_temp,
+	.sensors_count	= 2,
 };
 
 static int ltq_thermal_get_temp(void *data, int *temp)
@@ -244,6 +263,13 @@ static int ltq_thermal_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->chiptop);
 	}
 
+	priv->count = priv->tdata->sensors_count;
+	priv->sensors = devm_kzalloc(&pdev->dev, priv->count *
+				     sizeof(struct ltq_thermal_sensor),
+				     GFP_KERNEL);
+	if (!priv->sensors)
+		return -ENOMEM;
+
 	/* Init sensor */
 	priv->tdata->init(pdev, priv);
 
